### PR TITLE
fix: write secrets

### LINKED ISSUE #975: Some secrets updated by a user are not stored in values repo
**Describe the bug:** (a clear and concise description of what the bug is)

I observe app secrets edited by a user does not end up in the values repo (remote/origin) in Gitea

**To Reproduce**

1. Deploy Otomi with sops enabled
2. go to otomi console an edit prometheus (enable auth and feel password)
3. click Submit and then Deploy changes buttons
4. Observe that secret has not been changed in the Gitea repo

Note: This issue occurs if the secret file was not in repo in the first place.

**Expected behavior:** (a clear and concise description of what you expected to happen)
Updated but a user secrets are stored in Gitea repo

**Additional context:** (add any other context about the problem here)

Diving into the problem I noticed that after clicking Deploy Changes button:

Secret is stored in a .dec file in temporary user repo (e.g.: /tmp/otomi/values/bob.admin@otomi.cloud)
However, the secret is not passed to the root repo (e.g.: /tmp/otomi/values/main)
It happens because by design `.dec` files are not tracked by git

### PR DESCRIPTION
**Pull request by Morriz**
Writes secrets to correct file, core server now handles values write in `prepareValues`
(fixes #975)

depends-on: redkubes/otomi-api/pull/350
## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [ ] Helm releases are meeting otomi's baseline security policies, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
After make some changes in console submitting and clicking deploy changes the values are not saved anymore.

From logs it seems that writeValues function is missing some parameters
```
2022-12-12T10:51:32.206Z otomi:server:log Request to prepare values repo
2022-12-12T10:51:32.206Z otomi:common:values:writeValues:debug Writing values:  undefined
```

**Comment by Morriz** [comment]
@j-zimnowoda can you confirm that the tools server is running the correct tag? And api too?

**Comment by j-zimnowoda** [comment]
> @j-zimnowoda can you confirm that the tools server is running the correct tag? And api too?

yes, it was running the correct tools and api

**Review by staticvoid255 — APPROVED** [review]

**Comment by Morriz** [comment]
@j-zimnowoda can you confirm your observations are correct? Or give deeper insights? Why or what?

**Comment by j-zimnowoda** [comment]
> @j-zimnowoda can you confirm your observations are correct? Or give deeper insights? Why or what?

Maurice I have already describe the bug in https://github.com/redkubes/otomi-core/issues/975 and provided you another observation with logs and steps to reproduce. Could you please be more concrete and indicate what information is missing?

**Comment by Morriz** [comment]
@j-zimnowoda I added the fix 3 weeks ago. Can you confirm it works?

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Otomi does not install
```
operation: vendor/kes-to-eso/migrate-post.sh && echo "Success"
[162](https://github.com/redkubes/otomi-core/actions/runs/3836943442/jobs/6531683216#step:18:163)
2023-01-04T10:20:12.247Z ***global:error Error: /usr/bin/bash: vendor/kes-to-eso/migrate-post.sh: Permission denied
[163](https://github.com/redkubes/otomi-core/actions/runs/3836943442/jobs/6531683216#step:18:164)
    at execute (/home/app/stack/dist/src/cmd/upgrade.js:34:39)
[164](https://github.com/redkubes/otomi-core/actions/runs/3836943442/jobs/6531683216#step:18:165)
    exit code: 126 (Invoked command cannot execute)
[165](https://github.com/redkubes/otomi-core/actions/runs/3836943442/jobs/6531683216#step:18:166)
    at ChildProcess.<anonymous> (/home/app/stack/node_modules/zx/dist/bundle.cjs:15261:22)
[166](https://github.com/redkubes/otomi-core/actions/runs/3836943442/jobs/6531683216#step:18:167)
    at ChildProcess.emit (node:events:527:28)
[167](https://github.com/redkubes/otomi-core/actions/runs/3836943442/jobs/6531683216#step:18:168)
    at maybeClose (node:internal/child_process:1092:16)
[168](https://github.com/redkubes/otomi-core/actions/runs/3836943442/jobs/6531683216#step:18:169)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
```

File: charts/team-ns/templates/external-secrets.yaml [hunk 1]
```diff
@@ -2,41 +2,52 @@
...
+    {{- if $clusterWide }}
     matchLabels: 
       {}
-    {{- else }}
+    {{- else if $teamsWide }}
     matchExpressions:
       - {key: name, operator: In, values: [{{ $teamNamespacesCollection }}]}
+    {{- else if $hasNamespaces }}
+    matchExpressions:
+      - {key: name, operator: In, values: [{{ $namespacesSelection }}]}
```
Comment (j-zimnowoda) [inline charts/team-ns/templates/external-secrets.yaml:47]: each team each team has a common namespace label: `type: team`

**Comment by j-zimnowoda** [comment]
There is still an issue with writing values to files, which I reported before. 
I will continue investigating it.

File: charts/team-ns/templates/external-secrets.yaml (hunk 1 above)
Comment (Morriz) [reply charts/team-ns/templates/external-secrets.yaml:47]: of course...doh!

**Comment by Morriz** [comment]
> There is still an issue with writing values to files, which I reported before.
> I will continue investigating it.

solved with `app.use(json())` in server.

**Pull request closed without merging**